hledger-lib/Hledger/Reports/PostingsReport.hs

```diff
@@ -61,15 +65,44 @@ instance HasAmounts PostingsReportItem where
 -- with a tuple.
 type SummaryPosting = (Posting, Period)
 
+-- Possible value expressions taken by the --sort command
+-- Each of these takes a bool, which shows if it has been inverted
+-- (True -> has been inverted, reverse the order)
+data ValueExp
+    = Date' Bool
+    | Account' Bool
+    | Amount' Bool
+    deriving (Show)
+type SortSpec = [ValueExp]
+
+-- Load a SortSpec from the argument given to --sort
+-- If there is no spec given, then sort by [Date' False] by default
+getSortSpec :: RawOpts -> SortSpec
+getSortSpec opts = 
+    let opt = maybestringopt "sort" opts
+        optParser s = 
+          let terms = map strip $ splitOn "," s 
+              termParser t = case trimmed of
+                "date" -> Date' isNegated
+                "account" -> Account' isNegated
+                "amount" -> Amount' isNegated
+                _ -> error' $ "unsupported value expression '" ++ t ++ "' given to --sort"
+                where isNegated = isPrefixOf "-" t
+                      trimmed = fromMaybe t (stripPrefix "-" t)
+          in map termParser terms
+    in maybe [Date' False] optParser opt 
+
 -- | Select postings from the journal and add running balance and other
 -- information to make a postings report. Used by eg hledger's register command.
-postingsReport :: ReportSpec -> Journal -> PostingsReport
-postingsReport rspec@ReportSpec{_rsReportOpts=ropts@ReportOpts{..}} j = items
+postingsReport :: ReportSpec -> Maybe SortSpec -> Journal -> PostingsReport
```

Comment:
Similarly, I think passing `Nothing` or `Just sspec` to this is awkward. If the `SortSpec` was part of `ReportOpts` instead, then we could calculate the correct `sspec` inside of this function and not have to pollute other call sites.